### PR TITLE
feat: allow more than 1 lms configuration per lms-kind

### PR DESCRIPTION
**Pull request by johnnagro**
# Description

- added a uuid to canvas and blackboard configs for use in public urls
- use config-specific oauth state (uuid)
- change views to use new lookup, and fallback
- add oauth authorization url to canvas models

# References

- [ENT-5238](https://openedx.atlassian.net/browse/ENT-5238)

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/blackboard/migrations/0010_auto_20211221_1532.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Generated by Django 3.2.10 on 2021-12-21 15:32
+
+from django.db import migrations, models
+import uuid
+
+
+def create_uuid(apps, schema_editor):
```
Comment (binodpant) [inline integrated_channels/blackboard/migrations/0010_auto_20211221_1532.py:7]: do we need a reverse data migration to undo this for rollbacks?

File: integrated_channels/canvas/migrations/0013_auto_20211221_1535.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Generated by Django 3.2.10 on 2021-12-21 15:35
+
+from django.db import migrations, models
+import uuid
+
+
+def create_uuid(apps, schema_editor):
+    CanvasEnterpriseCustomerConfiguration = apps.get_model('canvas', 'CanvasEnterpriseCustomerConfiguration')
```
Comment (binodpant) [inline integrated_channels/canvas/migrations/0013_auto_20211221_1535.py:8]: ditto: any need for reverse functions?

**Review by binodpant — APPROVED** [review]
other than the minor nits I am good with this change... it does not impact the other channels. but we will have to keep this impl intact if we add more oauth cases so maybe in future iterations we can think of some common pattern. but totally not worth right now to DRY beyond this

File: integrated_channels/blackboard/migrations/0010_auto_20211221_1532.py (hunk 1 above)
Comment (johnnagro) [reply integrated_channels/blackboard/migrations/0010_auto_20211221_1532.py:7]: we can throw them away.

File: integrated_channels/canvas/migrations/0013_auto_20211221_1535.py (hunk 2 above)
Comment (johnnagro) [reply integrated_channels/canvas/migrations/0013_auto_20211221_1535.py:8]: we can throw them away.

**Pull request merged**